src/tools/MicroComGenerator/CppGen.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Linq;
+using System.Text;
+using MicroComGenerator.Ast;
+
+namespace MicroComGenerator
+{
+    public class CppGen
+    {
+        static string ConvertType(AstTypeNode type)
+        {
+            var name = type.Name;
+            if (name == "byte")
+                name = "unsigned char";
+            else if(name == "uint")
+                name = "unsigned int";
+            return name + new string('*', type.PointerLevel);
+        }
+        
+        public static string GenerateCpp(AstIdlNode idl)
+        {
+            var sb = new StringBuilder();
+            var preamble = idl.GetAttributeOrDefault("cpp-preamble");
+            if (preamble != null)
+                sb.AppendLine(preamble);
+
+            foreach (var s in idl.Structs)
+                sb.AppendLine("struct " + s.Name + ";");
```

Comment:
Split? Here and below.

```suggestion
            foreach (var s in idl.Structs)
            {
                sb.AppendLine("struct ");
                sb.AppendLine(s.Name);
                sb.AppendLine(";");
            }
```

Reply:
Well, I'm not really worried about efficiency of CPP codegen *that much* (it takes 6ms on my machine), StringBuilder is here mostly for readability. The Roslyn one, on the other hand, takes whole 2 seconds.

Reply:
![image](https://user-images.githubusercontent.com/1067584/97619764-aa6e3100-1a31-11eb-8a62-c82eb9347f3e.png)